packages/trpc/server/routers/viewer/eventTypes/update.handler.ts

```diff
@@ -245,26 +245,21 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
    * but the real detail will be inside app metadata, so with this you can have different prices in different apps
    * So the price and currency inside eventType will be deprecated soon or just keep as reference.
    */
-  if (
-    input.metadata?.apps?.alby?.price ||
-    input?.metadata?.apps?.paypal?.price ||
-    input?.metadata?.apps?.stripe?.price
-  ) {
-    data.price =
-      input.metadata?.apps?.alby?.price ||
-      input.metadata.apps.paypal?.price ||
-      input.metadata.apps.stripe?.price;
+
+  let enabledPaymentAppKey: string | undefined = undefined;
+
+  for (const appKey in input.metadata?.apps) {
+    const app = input.metadata?.apps[appKey];
+    if (app.enabled && app) {
+      enabledPaymentAppKey = appKey;
+      break;
+    }
   }
 
-  if (
-    input.metadata?.apps?.alby?.currency ||
-    input?.metadata?.apps?.paypal?.currency ||
-    input?.metadata?.apps?.stripe?.currency
-  ) {
-    data.currency =
-      input.metadata?.apps?.alby?.currency ||
-      input.metadata.apps.paypal?.currency ||
-      input.metadata.apps.stripe?.currency;
+  if (enabledPaymentAppKey) {
+    const paymentApp = input.metadata?.apps[enabledPaymentAppKey];
+    data.price = paymentApp.price;
+    data.currency = paymentApp.currency;
```

Comment:
this will create typescripts errors 
![Uploading Screenshot 2023-11-08 at 10.46.48.png…]()

Reply:
we should have utils in appstore to make this easier, something like this that we could generate 

```ts
 const PAYMENT_APP_KEYS: (keyof Pick<typeof appDataSchemas, "stripe" | "paypal" | "alby">)[] = [
    "paypal",
    "alby",
    "stripe",
  ];
  const eventTypePaymentApps = PAYMENT_APP_KEYS.map((appKey) => input.metadata?.apps?.[appKey]);
  const paymentApp = eventTypePaymentApps.find((app) => app?.enabled && app?.price);

  if (paymentApp) {
    data.price = paymentApp.price;
    data.currency = paymentApp.currency;
  }
```

Reply:
maybe @hariombalhara you have an idea to handle this ?

Reply:
There's a function called `getPaymentAppData` that exists. The thing is, it expects the entire return of `getEventTypeById` but it only looks at the event type metadata. Right now in the update handler we're not querying for the whole event type, just what we need.